### PR TITLE
Codegen

### PR DESCRIPTION
**Pull request by itsfuad**
This pull request introduces significant enhancements to the Ferret compiler, focusing on code generation, assembly output, and test coverage. Key changes include implementing assembly code generation, adding support for specifying output paths, and improving the test suite to validate new functionality. Below are the most important changes grouped by theme:

### Code Generation Enhancements
* Added `CompileToAssembly` function in `compiler/internal/backend/backend.go` to generate x86-64 assembly code from Ferret programs, with support for debug information and output file handling.
* Introduced assembly code generation logic in `compiler/cmd/main.go`, enabling the compiler to produce `.asm` files based on input programs and specified output paths.

### Output Path Support
* Updated `parseArgs` in `compiler/cmd/main.go` to handle a new `--output` flag for specifying custom assembly file paths.
* Enhanced test cases in `compiler/cmd/main_test.go` to validate the `--output` flag functionality and ensure correct handling of edge cases. [[1]](diffhunk://#diff-d1b78055f781c3820212b8782258a9c8d377de450e936c194006c16ab650ec26R41-R50) [[2]](diffhunk://#diff-d1b78055f781c3820212b8782258a9c8d377de450e936c194006c16ab650ec26L96-R134)

### Assembly Code Examples
* Added sample assembly files (`app/cmd/main.asm`, `app/test_import.asm`) showcasing variable declarations, arithmetic operations, and program structure for Ferret programs. [[1]](diffhunk://#diff-200c5b5b36c7cc720fd69eb464cfdfc7b24e1e56d3cf44b4866597b64d1b1f6fR1-R118) [[2]](diffhunk://#diff-bcf46c177ea7c39793f5c9dd00224a74e3b998249f740d5b39fb23942c4410f4R1-R38)

### Compiler Configuration
* Introduced `.ferret.json` configuration file to specify compiler settings, caching paths, and remote module dependencies.

### Miscellaneous Improvements
* Refactored `compiler/cmd/main.go` to include the `strings` package and streamline argument parsing logic.

### CODE REVIEW

File: compiler/cmd/main.go [hunk 1]
```diff
@@ -77,21 +78,46 @@ func Compile(filePath string, isDebugEnabled bool) *ctx.CompilerContext {
...
+
+	if isDebugEnabled {
+		colors.GREEN.Println("---------- [Code Generation done] ----------")
+	}
+
 	return context
 }
 
-func parseArgs() (string, bool, bool, string) {
+func parseArgs() (string, bool, bool, string, string) {
```
[the remark this answers is not in the document]
Comment (itsfuad) [reply compiler/cmd/main.go:102]: output path is checked on the Compile function
